src/devices/Mpu9250/Mpu9250.cs

```diff
@@ -0,0 +1,983 @@
+                if ((GyroscopeBandwidth != GyroscopeBandwidth.Bandwidth3600HzFS32) &&
+                    (GyroscopeBandwidth != GyroscopeBandwidth.Bandwidth8800HzFS32))
+                    return val / (1 + SampleRateDivider);
+                return val;
+            }
+        }
+
+        /// <summary>
+        /// Get the gyroscope in degrees per seconds
+        ///    +Z   +Y
+        ///  \  |  /
+        ///   \ | /
+        ///    \|/
+        ///    /|\
+        ///   / | \
+        ///  /  |  \
+        ///         +X
+        /// </summary>
+        public Vector3 Gyroscope => GetRawGyroscope() * GyroscopeConvertion;
+
+        private Vector3 GetRawGyroscope()
+        {
+            Span<byte> rawData = stackalloc byte[6] { 0, 0, 0, 0, 0, 0 };
+            Vector3 gyro = new Vector3();
+            ReadByteArray(Register.GYRO_XOUT_H, rawData);
+            gyro.X = BinaryPrimitives.ReadInt16BigEndian(rawData);
+            gyro.Y = BinaryPrimitives.ReadInt16BigEndian(rawData.Slice(2));
+            gyro.Z = BinaryPrimitives.ReadInt16BigEndian(rawData.Slice(4));
+            return gyro;
+        }
+
+        #endregion
+
+        #region Temperature
+
+        /// <summary>
+        /// Get the temperature
+        /// </summary>
+        public Temperature Temperature
+        {
+            get
+            {
+                Span<byte> rawData = stackalloc byte[2] { 0, 0 };
+                ReadByteArray(Register.TEMP_OUT_H, rawData);
+                // formula from the documentation
+                return Temperature.FromCelsius((BinaryPrimitives.ReadInt16BigEndian(rawData) - 21) / 333.87 + 21);
+            }
+        }
+
+        #endregion
+
+        #region Modes, constructor, Dispose
+
+        /// <summary>
+        /// Setup the Wake On Motion. This mode generate a rising signal on pin INT
+        /// You can catch it with a normal GPIO and place an interruption on it if supprted
+        /// Reading the sensor won't give any value until it wakes up preriodically
+        /// Only Accelerator data is availablein this mode
+        /// </summary>
+        /// <param name="magnetometerThreshold"></param>
+        /// <param name="acceleratorLowPower"></param>
+        public void SetWakeOnMotion(uint magnetometerThreshold, AccelerometerLowPowerFrequency acceleratorLowPower)
+        {
+            // Using documentation page 31 of Product Specification to setup
+            _wakeOnMotion = true;
+            if (magnetometerThreshold > 1020)
+                throw new ArgumentException($"{nameof(magnetometerThreshold)} has to be between 0mg and 1020mg");
+            // LSB = 4mg
+            magnetometerThreshold /= 4;
+            // We can't use the magnetometer, only Accelerometer will be measured
+            _ak8963.MeasurementMode = MeasurementMode.PowerDown;
+            // Make sure we start from a clean soft reset
+            PowerOn();
```

Comment:
Do we need to call this again? I ask since we are calling powerOn on the ctor so presumably the device will already be on.

Reply:
Yes, it is necessary if the device has been put in another mode. It's a precaution. In 99% of the cases, this function will be called at the correct moment, at the beginning without changing anything to the device. Now, if not done properly, better to make sure all is in order.